### DISCUSSION TITLE
Is there a way to generate SVG in the production build

### DISCUSSION THREAD

**Discussion by lidonis** [nuxt/nuxt#9676] [Questions (Nuxt 2)] [answered]
I want to be able to have a Vue component to generate SVG using JavaScript and include it with parameters in several pages
For now I have put all my generation code in the _mounted_ function but with that the SVG is generated client side.

I'd like to have it embedded in the render HTML of the file generated a build time to reduce loading time in the browser

**Comment by kissu** [comment]
You could maybe generate it in [`created`](https://nuxtjs.org/docs/2.x/concepts/nuxt-lifecycle#lifecycle`) since this is available in both server and client.

**Reply by lidonis** [reply]
Unfortunately  I don't have access to the DOM at the created stage. I need it to create SVG elements.

**Comment by TheAlexLichter** [comment]
Could you share a small repo of what you are trying to achieve?

**Reply by lidonis** [reply]
Here it is https://github.com/lidonis/nuxt-ssg-svg

**Reply by TheAlexLichter** [reply]
Ah, I see.
As `rough.js` needs the `document` to create SVGs, it cannot be used server-side easily (see also https://github.com/rough-stuff/rough/issues/169 for people who tried using it with pure Node.js).
You can either think about a "fake dom" like xmldom (or jsdom?) and ensure this lib is only used on the server-side. Major downside: You have to add logic for client and server-side separately.

TL;DR - not easily achievable but likely possible.

**Comment by lidonis** [comment, marked as answer]
I found a way around my problem using rough.js generator. With that I am able to draw my SVG and have it in the generated HTML file.
See the result on this branch https://github.com/lidonis/nuxt-ssg-svg/tree/rough_generator

Thanks everyone for helping
